docs/kubernetes_auth.md

```diff
@@ -0,0 +1,133 @@
+# Authenticating with Kubernetes
+
+In order to connect to your Kubernetes cluster, you'll need to set the environment
+variable `KUBECONFIG_DATA`. This variable should contain a valid base64 encoded
+[kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/)
+
+## Quickstart
+> Note: Using your local kubeconfig is NOT recommended. It is much more secure
+> to create a service account, as shown below
+
+The easiest way to authenticate is to pass a valid kubeconfig to the `base64` command.
+
+```
+cat ~/.kube/config | base64 -w 0
+```
+
+The output of this command should be set as the environment variable `KUBECONFIG_DATA`
+in the settings for your CI platform.
+
+## Using Cloud Providers
+rok8s-scripts can use the `aws-cli` and `gcloud` tools to authenticate with cloud providers.
+See the documentation for [AWS](aws.md) or [GCP](gcp.md) for more information.
+
+## Using a Service Account
+[Service accounts](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/)
```

Comment:
I would say let's put the "preferred method" as using your cloud provider IAM.  This is the second-best way.  Especially if you are in GCP.  Otherwise, I love the addition of this doc and the guidance it gives.

Reply:
👍 done